src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate.swift

```diff
@@ -322,11 +324,34 @@ class AppDelegate: UIResponder, UIApplicationDelegate, CoronaWarnAppDelegate, Re
 		return VaccinationValueSetsProvider(client: CachingHTTPClient(), store: store)
 	}
 	
-	private lazy var healthCertificateValidationService: HealthCertificateValidationProviding = HealthCertificateValidationService(
-		store: store,
-		client: client,
-		vaccinationValueSetsProvider: vaccinationValueSetsProvider
-	)
+	
+	private lazy var healthCertificateValidationService: HealthCertificateValidationProviding = {
+		#if DEBUG
+		if isUITesting {
+			var mock = MockHealthCertificateValidationService()
+			
+			if LaunchArguments.healthCertificate.invalidCertificateCheck.boolValue {
+				
+				// Provide data for invalid validation
+				let fakeResult: ValidationResult = .fake(result: .fail)
+				fakeResult.rule?.description = [Description(lang: "de", desc: "Die Impfreihe muss vollständig sein (z.B. 1/1, 2/2)."),
+												Description(lang: "en", desc: "The vaccination schedule must be complete (e.g., 1/1, 2/2).")]
+				mock.validationResult = .success(.validationFailed([fakeResult]))
+			} else {
+				mock.validationResult = .success(.validationPassed([.fake(), .fake(), .fake()]))
+			}
+			
+			return mock
+		}
+		#endif
+		
+		let service = HealthCertificateValidationService(
+			store: store,
+			client: client,
+			vaccinationValueSetsProvider: vaccinationValueSetsProvider
+		)
+		return service
```

Comment:
```suggestion
		return HealthCertificateValidationService(
			store: store,
			client: client,
			vaccinationValueSetsProvider: vaccinationValueSetsProvider
		)
```

Just to make it a little simpler 🤓